superset/advanced_data_type/plugins/asn.py

```diff
@@ -0,0 +1,50 @@
+import re
+from superset.advanced_data_type.plugins import translate_filter_func
+from superset.advanced_data_type.plugins.operator_sets import EQUAL_NULLABLE_OPERATOR_SET, PATTERN_MATCHING_OPERATOR_SET
+from superset.advanced_data_type.plugins.utils.safely_get_int_value import safely_get_int_value
+from superset.advanced_data_type.types import AdvancedDataType, AdvancedDataTypeRequest, AdvancedDataTypeResponse
+
+MIN_ASN_16B = 1
+MAX_ASN_16B = 65534
+MIN_ASN_32B = 131072
+MAX_ASN_32B = 4294967294
+
+def asn_func(req: AdvancedDataTypeRequest) -> AdvancedDataTypeResponse:
+    """
+    Convert a passed in AdvancedDataTypeRequest to an AdvancedDataTypeResponse
+    """
+    resp: AdvancedDataTypeResponse = {
+        "values": [],
+        "error_message": "",
+        "display_value": "",
+        "valid_filter_operators": EQUAL_NULLABLE_OPERATOR_SET,
+    }
+    if req["values"] == [""]:
+        resp["error_message"] = "ASN must not be empty"
+        return resp
+    for val in req["values"]:
+        string_value = str(val)
+        """
+        Seemed easier to just check the range given it can only be between certain values.
+        https://www.cloudflare.com/learning/network-layer/what-is-an-autonomous-system/#What%20Is%20An%20Autonomous%20System%20Number%20(ASN)?
+        Note that sometimes they are presented as AS(num), but existing ASNs on Superset are just numbers
+        so we will keep it that way for consistency.
+        """
+        int_value = safely_get_int_value(string_value)
```

Comment:
This function returns 0 if the string is not a valid number. I don't think we should use it in this case because it will effectively make an invalid value valid.

Reply:
sure, I made it return -1.

Reply:
I'm still not sure about this value. Why do we want to return a bogus value that could break things further down the line? Why not simply handle the `ValueError` right here? Passing an invalid number here is absolutely a validation error. Why are we fudging it instead of letting the user know?
Also, I think this is the only place that function is ever used - can we just remove the function?

Reply:
Okay, I hear you. I'll adjust the logic to just handle the error.

Reply:
c59a750c4170ec83add4cd7d7d6eba5eb5ae3007